File: problems/002/p002.php
<?php

function resolve($max): int
{
    $x = 1;
    $y = 2;
    $sum = $y;

    while ($y < $max) {
        $x1 = $y;
        $y2 = $x + $y;

        if ($y2 % 2 == 0) {
            $sum += $y2;
        }

        $x = $x1;
        $y = $y2;
    }

    return $sum;
}

function main()
{
    $result = resolve(4000000);

    echo "Result=$result\n";
}

main();

File: problems/002/p002.c
#include <stdio.h>

int resolve(int max)
{
	int x = 1;
	int y = 2;
	int sum = y;
	int temp;

	while (y < max)
	{
		temp = y;
		y = x + y;
		x = temp;

		if (y % 2 == 0)
		{
			sum = sum + y;
		}
	}
	return sum;
}

int main()
{

	int result = resolve(4000000);

	printf("Result=%d\n", result);

	return 0;
}
